cranelift-frontend/src/frontend.rs

```diff
@@ -41,8 +41,17 @@ pub struct FunctionBuilder<'a> {
 
 #[derive(Clone, Default)]
 struct EbbData {
-    filled: bool,
+    /// An Ebb is "pristine" iff no instructions have been added since the last
+    /// call to `switch_to_block()`.
     pristine: bool,
+
+    /// An Ebb is "filled" iff a terminator instruction has been inserted since
+    /// the last call to `switch_to_block()`.
+    ///
+    /// A filled block cannot be pristine.
+    filled: bool,
```

Comment:
follow-up:
If a filled block cannot be pristine, it sound to me that these 2 flag can represent a state which is not allowed to exists.
Shouldn't an `enum` better describe this state.

```rust
enum EbbState {
  UnChanged,
  Changed,
  ChangedAndTerminated,
}
```

Reply:
That's a great observation! I'll look into that. I suspect the enum would work and be clearer.

Reply:
`is_pristine()` and `is_filled()` appears to be part of the public API for `cranelift-frontend`, so cleaning that up is a bit of a bigger thing. I'll file an issue for tracking.